src/Controls/src/Core/Platform/ModalNavigationManager/ModalNavigationManager.Android.cs

```diff
@@ -356,11 +368,18 @@ public override void OnStart()
 				var dialog = Dialog;
 
 				if (dialog is null || dialog.Window is null || View is null)
+				{
+					// SAFETY: Fire event even on early return to prevent deadlock
+					FirePresentationCompleted();
```

Comment:
Don't believe we should Fire here, since `OnStart` is too early for navigation. Also, if this check results in `true` something is really wrong and should fail.

Reply:
@pictos, I have added FirePresentationCompleted() in the null check condition to prevent potential deadlocks as per @PureWeen's suggestion. Please find the below comment for your reference.

Comment link: https://github.com/dotnet/maui/pull/32479#pullrequestreview-3500799115

Reply:
@BagavathiPerumal I read it, but not sure if we just want to complete the Task or set an exception to the user, since the navigation will be in a invalid state.

cc: @PureWeen